module/zfs/dnode_sync.c

```diff
@@ -207,9 +207,7 @@ free_verify(dmu_buf_impl_t *db, uint64_t start, uint64_t end, dmu_tx_t *tx)
 			continue;
 		ASSERT(err == 0);
 		ASSERT(child->db_level == 0);
-		dr = child->db_last_dirty;
-		while (dr && dr->dr_txg > txg)
-			dr = dr->dr_next;
+		dr = dbuf_find_dirty_lte(child, txg);
 		ASSERT(dr == NULL || dr->dr_txg == txg);
```

Comment:
looks like this could also use `dbuf_find_dirty_eq`

Reply:
That's what I was using initially. However, you lose the ability to assert that there are no transactions earlier than txg in the list. 

In other words:
```
 		ASSERT(dr == NULL || dr->dr_txg == txg); 
```
The second half of the assertion becomes a no-op. If this is ok, I can convert it back and remove the assert.